implementation/src/main/java/com/teampc/controller/ViewSubmissionsController.java

```diff
@@ -57,32 +60,17 @@ public void setCurrentTest(Test test) {
       className.setText(currentTest.getCourseName());
 
       Collection<Submission> submissions = SubmissionDAO.getInstance().fetchSubmissionsForTest(currentTest);
-      Key key = currentTest.getKey();
-
-      // CANNED DATA
-      if (submissions.isEmpty()) {
-         Submission s = new Submission();
-         s.responses.add(new MultipleChoiceQuestionResponse());
-         s.responses.add(new MultipleChoiceQuestionResponse());
-         s.responses.add(new ShortAnswerQuestionResponse("", ShortAnswerQuestionResponse.MatchType.ALL));
-         s.responses.add(new ShortAnswerQuestionResponse("", ShortAnswerQuestionResponse.MatchType.ALL));
-         s.taker = UserSession.getLoggedInUser();
-         s.setTest(currentTest);
-
-         submissions = new ArrayList<>();
-         submissions.add(s);
-
-         key = new Key();
-         key.responses.add(new MultipleChoiceQuestionResponse());
-         key.responses.add(new MultipleChoiceQuestionResponse(1, new ArrayList<>()));
-         key.responses.add(new ShortAnswerQuestionResponse("", ShortAnswerQuestionResponse.MatchType.ALL));
-         key.responses.add(new ShortAnswerQuestionResponse("one", ShortAnswerQuestionResponse.MatchType.ALL));
-      }
+      Key key = currentTest.generateKey();
 
-      final Key keyPC = key;
-      submissions.stream().forEach(nextSubmission -> nextSubmission.gradeTest(keyPC));
+      if (key == null) {
+         LOG.error("could not get key");
+      }
+      else {
+         final Key keyPC = key;
+         submissions.stream().forEach(nextSubmission -> nextSubmission.gradeTest(keyPC));
 
-      submissions.forEach(submission -> data.add(new RowView<>(new SubmissionRowController(submission, currentTest), RESOURCE)));
+         submissions.forEach(submission -> data.add(new RowView<>(new SubmissionRowController(submission, currentTest), RESOURCE)));
```

Comment:
should probably combine this forEach w/ the previous call, only make 1 pass through the data